plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -118,7 +119,7 @@ fu_thunderbolt_device_get_version(FuThunderboltDevice *self, GError **error)
 		return FALSE;
 	}
 
-	for (guint i = 0; i < 50; i++) {
+	for (guint i = 0; i < TBT_NVM_RETRY_COUNT; i++) {
```

Comment:
@Westeri, was there some history here I'm forgetting?  I seem to recall there was some reason for it being this big.

Reply:
Maybe it was from TBT3 devices that didn't support RTD3 and we were waiting for BIOS to populate the topology or something?  If so; I guess I'd be more comfortable if we the timeout was only made smaller for USB4 devices.

Reply:
Good question. One thing that might explain it is that when the NVM authentication (firmware upgrade) is started the router gets hot-removed and it may take some time for it come back. Not sure it matters here  though. I would think the original device gets removed by uevent in fwupd too.

Reply:
At least in this context that shouldn't matter (we have separate code that waits for devices to come back and this function isn't called until they're back).  I did some history analysis and it looks like this specific retry came in b9ac0aa179, but it was copying existing behavior.  Looking further back, I first introduced it with 25d51d14297713b4817ef9644f83e07e71344b0a because of #608 and we discussed it in #609.

It was first seen in https://bugs.launchpad.net/ubuntu/+source/fwupd/+bug/1768627, which looks to be a pretty old FW CM system.  IMO I'd say at this point we should be safe to decrease the retry count on anything USB4.

Reply:
I've split out a w/a into https://github.com/fwupd/fwupd/pull/8213

Reply:
How much if any of **this** branch still makes sense?

Reply:
Can you rebase it and see?

Reply:
Done.